tests/fuzz/s2n_openssl_diff_pem_parsing_test.c

```diff
@@ -81,35 +81,39 @@ static int openssl_parse_cert_chain(struct s2n_stuffer *in)
 
 }
 
-static int s2n_parse_cert_chain(struct s2n_stuffer *in)
+static int s2n_parse_cert_chain(struct s2n_stuffer *in, uint8_t *chain_len)
 {
     struct s2n_cert_chain_and_key *chain_and_key = s2n_cert_chain_and_key_new();
+    *chain_len = 0;
     
     /* Allocate the memory for the chain and key */
-    s2n_create_cert_chain_from_stuffer(chain_and_key->cert_chain, in);
+    if (s2n_create_cert_chain_from_stuffer(chain_and_key->cert_chain, in) != S2N_SUCCESS) {
+        s2n_cert_chain_and_key_free(chain_and_key);
+        S2N_ERROR_PRESERVE_ERRNO();
+    }
 
     struct s2n_cert *next = chain_and_key->cert_chain->head;
-    int chain_len = 0;
     while(next != NULL) {
-        chain_len++;
+        *chain_len += 1;
         next = next->next;
     }
-    
+
     s2n_cert_chain_and_key_free(chain_and_key);
-    return chain_len;
+
+    return 0;
 }
 
 int LLVMFuzzerTestOneInput(const uint8_t *buf, size_t len)
 {
-    struct s2n_stuffer in;
+    struct s2n_stuffer in = {0};
     GUARD(s2n_stuffer_alloc(&in, len + 1));
     GUARD(s2n_stuffer_write_bytes(&in, buf, len));
     in.blob.data[len] = 0;
 
     uint8_t openssl_chain_len = openssl_parse_cert_chain(&in);
     GUARD(s2n_stuffer_reread(&in));
-    uint8_t s2n_chain_len = s2n_parse_cert_chain(&in);
-    GUARD(s2n_stuffer_free(&in));
+    uint8_t s2n_chain_len;
+    s2n_parse_cert_chain(&in, &s2n_chain_len);
```

Comment:
Should there be a GUARD here?

Reply:
Yes, here and in another location in s2n_parse_cert_chain (added both).